curriculum/challenges/english/blocks/workshop-shortest-path-algorithm/6876089f7a500b7c782e5b83.md

```diff
@@ -0,0 +1,64 @@
+---
+id: 6876089f7a500b7c782e5b83
+title: Step 13
+challengeType: 20
+dashedName: step-13
+---
+
+# --description--
+
+You need to decide whether the current node is a better choice than the one you've already found (if any). To do this, you’ll add a conditional statement inside the loop.
+
+The condition should do two things:
+
+- Check if the node has not been visited yet.
+- Compare the known distance to the current `min_distance`.
+
+Inside the `for node_no in range(n):` loop, add an `if` statement that checks whether `node_no` has not been visited **and** whether `distances[node_no]` is less than `min_distance`. 
+
+Add `pass` as a placeholder inside the conditional block.
+
+# --hints--
+
+You should add an `if` statement that checks if the node is unvisited and has a shorter distance.
+
+```js
+({
+  test: () => {
+    const code = runPython(`_code`);
+    assert(code.includes("if not visited[node_no] and distances[node_no] < min_distance:"));
+  }
+})
+```
```

Comment:
````suggestion
You should add an `if` statement inside your inner `for` loop.

```js
({
  test: () => {
    assert(runPython(`_Node(_code).find_function("shortest_path").find_for_loops()[0].find_bodies()[0].find_for_loops()[0].find_bodies()[0].find_ifs()[0]`))
  }
})
```

Your `if` statement should check if the node is unvisited (`visited[node_no]` is falsy) and `distances[node_no]` is less than `min_distance`.

```js
({
  test: () => {
    assert(runPython(`_Node(_code).find_function("shortest_path").find_for_loops()[0].find_bodies()[0].find_for_loops()[0].find_bodies()[0].find_ifs()[0].find_conditions()[0].is_equivalent("not visited[node_no] and distances[node_no] < min_distance")`))
  }
})
```
````